wikimedia/wikipedia-ios
# Add empty Notifications Center

**Phabricator:** https://phabricator.wikimedia.org/T288654

### Notes
Adds empty Notifications Center view to app, displayed when tapping the Notifications Bell in either the Explore feed or Settings when Explore is off.

- Adds empty `UICollectionView`
- Adds empty `NotificationsCenterViewModel` to transit data/`RemoteNotificationsController` to view controller
- Modifies `ViewControllerTransitionsController` to support standard push/pop animation style
   - [This approach](https://github.com/wikimedia/wikipedia-ios/blob/30dc4fcc59603d8cd8686b654753d632b06a0b7d/Wikipedia/Code/ViewControllerTransitionsController.swift#L110
) may be slightly controversial. One might argue it'd be better to just have a property on the `ViewController` base class to flag this desired behavior, but I think that would create an unclear expectation of what that property does. Truly, we're just wanting to avoid the transition style for `ViewController`s under the influence of `ViewControllerTransitionsController` – not all `ViewController`s period. So while this may look a little odd, I think it focuses behavioral expectations better than adding a `ViewController` bool.
- Adds `NotificationsCenterPresentationDelegate` to allow `WMFAppViewController` to handle conditional presentation logic of the Notifications Center based on the origin `ViewController`. I'd imagine keeping this presentation logic in `WMFAppViewController` makes sense, as this is where we'll likely want to handle and conditionally present the Notifications Center on incoming deep link taps of iOS push notifications as well.

### Test Steps
1. Confirm Notifications Center displays in Explore feed and Settings when Explore is off when tapping the Notifications Bell button
2. Confirm it displays in either device screen orientation
3. Ensure its appearance and disappearance animations are the standard navigation push/pop style
4. Ensure other Explore feed animation transitions (like tapping a featured article) perform as they did before

### Screenshots
![Notifications Center](https://user-images.githubusercontent.com/5652327/130833155-fbdbb739-d444-4d99-ab33-387e4a44e40b.png)
